Feature/add trait columns

zsloan opened the pull request:
This PR adds the option to add traits from a collection as extra pseudo-"case attribute" columns on the trait page.

The user can then filter off of those columns just like off of normal case attribute columns

zsloan committed:
Create button for adding traits to table

zsloan committed:
Destroy/empty table before creation

This is necessary to clear new column headers, but I need to change it
so that it only does it when adding new columns

zsloan committed:
Add JS for adding traits to table

zsloan committed:
Only empty table container on loading if data is loaded from JSON

Otherwise the row data is effectively deleted. This was initially done
because it appears to be the only way to add columns to an existing data
table that uses Scroller.

zsloan committed:
Make Transform/Filter options work with added trait columns

This also changes the way columns are added; previously a list of new
columns was passed to the table generation code, but now I just add them
to the attribute data in js_data

The main remaining feature to add is the ability to remove columns
(which I'll probably do as just the ability to remove all columns; I'm
not sure how to reasonable add the option to remove specific columns)

robwwilliams commented:
Very cool. Excited to test. Thanks.

On Thu, Nov 17, 2022 at 3:57 PM zsloan ***@***.***> wrote:

> This PR adds the option to add traits from a collection as extra
> pseudo-"case attribute" columns on the trait page.
>
> The user can then filter off of those columns just like off of normal case
> attribute columns
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/genenetwork/genenetwork2/pull/748
> Commit Summary
>
>    - 7aa792b
>    <https://github.com/genenetwork/genenetwork2/pull/748/commits/7aa792b0616fab1268269ad8c4675117c42544ad>
>    Create button for adding traits to table
>    - e710c91
>    <https://github.com/genenetwork/genenetwork2/pull/748/commits/e710c917aea735f2e56e28dd31aeabcd7fa5b1de>
>    Destroy/empty table before creation
>    - 6651900
>    <https://github.com/genenetwork/genenetwork2/pull/748/commits/6651900639e9d02af999a4cefe425fd0415285e1>
>    Add JS for adding traits to table
>    - 8995c0c
>    <https://github.com/genenetwork/genenetwork2/pull/748/commits/8995c0c9891cea962eadbdd868ac9705862a479f>
>    Only empty table container on loading if data is loaded from JSON
>    - 1db2834
>    <https://github.com/genenetwork/genenetwork2/pull/748/commits/1db28340ee703051645365a50eb10c9c25b5aebb>
>    Make Transform/Filter options work with added trait columns
>
> File Changes
>
> (5 files <https://github.com/genenetwork/genenetwork2/pull/748/files>)
>
>    - *A* wqflask/wqflask/static/new/javascript/add_traits_to_table.js
>    <https://github.com/genenetwork/genenetwork2/pull/748/files#diff-d0479702a81593994d84e2bf303e921dbf8627fa9c9970933fc14c9ba9d0eabc>
>    (300)
>    - *M* wqflask/wqflask/static/new/javascript/create_datatable.js
>    <https://github.com/genenetwork/genenetwork2/pull/748/files#diff-1393bb3fb2ef209a1982ade89b132b1fb97ebf0f80dc37202dd40ad37249a30a>
>    (5)
>    - *M*
>    wqflask/wqflask/static/new/javascript/initialize_show_trait_tables.js
>    <https://github.com/genenetwork/genenetwork2/pull/748/files#diff-09ae307d0488a7dc7c9eedd7c6e897e363857f650a6cb73933f6e1a8cb6e1f8a>
>    (331)
>    - *M* wqflask/wqflask/static/new/javascript/show_trait.js
>    <https://github.com/genenetwork/genenetwork2/pull/748/files#diff-3edfd5623c8d344941dcfe7356a4d540334f4c6c67ff97a47a82e6c1f8abd9f9>
>    (11)
>    - *M* wqflask/wqflask/templates/show_trait_edit_data.html
>    <https://github.com/genenetwork/genenetwork2/pull/748/files#diff-d63ef77225a78bad48e2b78087202866123fa9a10ae112a7e660079520c6b4ec>
>    (2)
>
> Patch Links:
>
>    - https://github.com/genenetwork/genenetwork2/pull/748.patch
>    - https://github.com/genenetwork/genenetwork2/pull/748.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/748>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7HJLBVUGYHOYGQYAGLWI2S3XANCNFSM6AAAAAASD3B3NU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Rob

Robert W Williams PhD
Department of Genetics, Genomics and Informatics
71 S Manassas St, Memphis TN 38163
University of Tennessee Health Science Center, TSRB 405
 Cell: 901 604 4752
EMAIL: ***@***.***
Zoom: https://tennesseehipaa.zoom.us/j/9492269658
----------------

zsloan commented:
It isn't on production yet, but it is on my branch right now.

The option to remove added columns doesn't exist yet (and when I do add it,
it'll probably just be the option to remove all added columns, since doing
it for specific ones will greatly complicate things), but the transform
functions should work.

One complication (that shouldn't be hard to address, but need to decide
how) is the existence of x values. Currently it just completely ignores all
samples like that. Probably easier to just show what I'm talking about, so
I'll do that tomorrow morning.

On Thu, Nov 17, 2022, 4:12 PM Rob Williams ***@***.***> wrote:

> Very cool. Excited to test. Thanks.
>
> On Thu, Nov 17, 2022 at 3:57 PM zsloan ***@***.***> wrote:
>
> > This PR adds the option to add traits from a collection as extra
> > pseudo-"case attribute" columns on the trait page.
> >
> > The user can then filter off of those columns just like off of normal
> case
> > attribute columns
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/genenetwork/genenetwork2/pull/748
> > Commit Summary
> >
> > - 7aa792b
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/commits/7aa792b0616fab1268269ad8c4675117c42544ad
> >
> > Create button for adding traits to table
> > - e710c91
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/commits/e710c917aea735f2e56e28dd31aeabcd7fa5b1de
> >
> > Destroy/empty table before creation
> > - 6651900
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/commits/6651900639e9d02af999a4cefe425fd0415285e1
> >
> > Add JS for adding traits to table
> > - 8995c0c
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/commits/8995c0c9891cea962eadbdd868ac9705862a479f
> >
> > Only empty table container on loading if data is loaded from JSON
> > - 1db2834
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/commits/1db28340ee703051645365a50eb10c9c25b5aebb
> >
> > Make Transform/Filter options work with added trait columns
> >
> > File Changes
> >
> > (5 files <https://github.com/genenetwork/genenetwork2/pull/748/files>)
> >
> > - *A* wqflask/wqflask/static/new/javascript/add_traits_to_table.js
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/files#diff-d0479702a81593994d84e2bf303e921dbf8627fa9c9970933fc14c9ba9d0eabc
> >
> > (300)
> > - *M* wqflask/wqflask/static/new/javascript/create_datatable.js
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/files#diff-1393bb3fb2ef209a1982ade89b132b1fb97ebf0f80dc37202dd40ad37249a30a
> >
> > (5)
> > - *M*
> > wqflask/wqflask/static/new/javascript/initialize_show_trait_tables.js
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/files#diff-09ae307d0488a7dc7c9eedd7c6e897e363857f650a6cb73933f6e1a8cb6e1f8a
> >
> > (331)
> > - *M* wqflask/wqflask/static/new/javascript/show_trait.js
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/files#diff-3edfd5623c8d344941dcfe7356a4d540334f4c6c67ff97a47a82e6c1f8abd9f9
> >
> > (11)
> > - *M* wqflask/wqflask/templates/show_trait_edit_data.html
> > <
> https://github.com/genenetwork/genenetwork2/pull/748/files#diff-d63ef77225a78bad48e2b78087202866123fa9a10ae112a7e660079520c6b4ec
> >
> > (2)
> >
> > Patch Links:
> >
> > - https://github.com/genenetwork/genenetwork2/pull/748.patch
> > - https://github.com/genenetwork/genenetwork2/pull/748.diff
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/genenetwork/genenetwork2/pull/748>, or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AC55V7HJLBVUGYHOYGQYAGLWI2S3XANCNFSM6AAAAAASD3B3NU
> >
> > .
> > You are receiving this because you are subscribed to this thread.Message
> > ID: ***@***.***>
> >
>
>
> --
> Rob
>
> Robert W Williams PhD
> Department of Genetics, Genomics and Informatics
> 71 S Manassas St, Memphis TN 38163
> University of Tennessee Health Science Center, TSRB 405
> Cell: 901 604 4752
> EMAIL: ***@***.***
> Zoom: https://tennesseehipaa.zoom.us/j/9492269658
> ----------------
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/748#issuecomment-1319275903>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGCMMZD2VWMCO7PHCITWI2UWRANCNFSM6AAAAAASD3B3NU>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

zsloan committed:
Fix remaining issues from rebase

zsloan committed:
Add add_traits_to_table.js

This adds the JS that handles adding addition columns to the trait page table for any traits a user selections from their collections

zsloan committed:
Fix the way table width is recalculated after adding trait columns

zsloan committed:
Round added trait column values to 3 decimal places

zsloan committed:
Fix issue where added trait columns were overwriting the last table
column

zsloan committed:
Remove unused empty_tables function
